### PR TITLE
Study updates

### PR DESCRIPTION
**Pull request by alxdarksage**
- added a number of fields to Study object, including a collection of Contact objects
- very minimal enforcement of phase: existing studies have "legacy," new studies are in "design," and you can't currently change this through an update;
- added some fields to the summary version of a study (createdOn, modifiedOn, phase);
- remove the study persistence exception handler and use the more robust MySQL exception handler, moving over the foreign key constraint and the tests for it.

### CODE REVIEW

File: src/main/java/org/sagebionetworks/bridge/hibernate/HibernateHelper.java [hunk 1]
```diff
@@ -40,11 +40,11 @@ public HibernateHelper(SessionFactory hibernateSessionFactory, PersistenceExcept
...
     public <T> void deleteById(Class<T> clazz, Serializable id) {
+        T obj = getById(clazz, id);
         // Hibernate optimistic versioning also applies to deletes. However, unlike updates, when we delete something,
         // we want it gone, so we generally don't care about optimistic versioning. In order to handle this in
         // Hibernate, we need to load the whole object before deleting it.
-        executeWithExceptionHandling(null, session -> {
-            T obj = session.get(clazz, id);
+        executeWithExceptionHandling(obj, session -> {
             session.delete(obj);
             return null;
```
Comment (alxdarksage) [inline src/main/java/org/sagebionetworks/bridge/hibernate/HibernateHelper.java:49]: Did you write a test for this?

File: src/main/java/org/sagebionetworks/bridge/validators/StudyValidator.java [hunk 2]
```diff
@@ -21,15 +27,37 @@ public void validate(Object object, Errors errors) {
         Study study = (Study)object;
         
         if (isBlank(study.getIdentifier())) {
-            errors.rejectValue("id", "is required");
-        } else if (!study.getIdentifier().matches(BridgeConstants.BRIDGE_EVENT_ID_PATTERN)) {
-            errors.rejectValue("id", BridgeConstants.BRIDGE_EVENT_ID_ERROR);
+            errors.rejectValue("identifier", "is required");
+        } else if (!study.getIdentifier().matches(BRIDGE_EVENT_ID_PATTERN)) {
+            errors.rejectValue("identifier", BRIDGE_EVENT_ID_ERROR);
```
Comment (alxdarksage) [inline src/main/java/org/sagebionetworks/bridge/validators/StudyValidator.java:32]: Make sure BSM is showing error now that identifier has changed name!

File: src/test/java/org/sagebionetworks/bridge/services/StudyServiceTest.java [hunk 3]
```diff
@@ -320,13 +326,16 @@ public void updateStudy() {
...
         assertNotNull(returnedValue.getModifiedOn());
     }
     
-    @Test(expectedExceptions = InvalidEntityException.class)
+    @Test(expectedExceptions = EntityNotFoundException.class)
     public void updateStudyInvalidStudy() {
-        service.updateStudy(TEST_APP_ID, Study.create());
+        Study study = Study.create();
+        study.setIdentifier("oneId");
+        service.updateStudy(TEST_APP_ID, study);
```
Comment (alxdarksage) [inline src/test/java/org/sagebionetworks/bridge/services/StudyServiceTest.java:338]: So, why did this change? Is this correct?

File: src/main/java/org/sagebionetworks/bridge/hibernate/HibernateHelper.java (hunk 1 above)
Comment (alxdarksage) [reply src/main/java/org/sagebionetworks/bridge/hibernate/HibernateHelper.java:49]: Added

File: src/main/java/org/sagebionetworks/bridge/validators/StudyValidator.java (hunk 2 above)
Comment (alxdarksage) [reply src/main/java/org/sagebionetworks/bridge/validators/StudyValidator.java:32]: Yes, it does.

File: src/test/java/org/sagebionetworks/bridge/services/StudyServiceTest.java (hunk 3 above)
Comment (alxdarksage) [reply src/test/java/org/sagebionetworks/bridge/services/StudyServiceTest.java:338]: Phase is required, so to ensure this doesn't trigger an NPE, we set it from the existing object before validation. But this requires having an identifier, which we virtually always will now that the controller sets it from the path (which is our default). So the ID is required and the exception thrown has changed.

**Review by nategbrown9 — APPROVED** [review]
Looks good.

**Pull request merged**